tracee-ebpf/main.go

```diff
@@ -67,11 +67,48 @@ func main() {
 			if c.Bool("security-alerts") {
 				cfg.Filter.EventsToTrace = append(cfg.Filter.EventsToTrace, tracee.MemProtAlertEventID)
 			}
-			bpfFile, err := getBPFObject()
+
+			// Check if user specified a bpf object path explicitly
+			userSpecifiedObjectPath, err := checkTraceeBPFEnvPath()
 			if err != nil {
 				return err
 			}
-			cfg.BPFObjPath = bpfFile
+			if userSpecifiedObjectPath != "" {
+				if debug {
+					fmt.Printf("BPF object file specified by TRACEE_BPF_FILE found: %s", userSpecifiedObjectPath)
+				}
+				cfg.BPFObjPath = userSpecifiedObjectPath
+				bpfBytes, err := ioutil.ReadFile(userSpecifiedObjectPath)
+				if err != nil {
+					return err
+				}
+				cfg.BPFObjBytes = bpfBytes
+			} else if btfEnabled() {
+				if debug {
+					fmt.Println("BTF enabled, attempting to unpack CORE bpf object")
+				}
+				cfg.BPFObjPath = "embed-core"
+				bpfBytes, err := unpackCOREBinary()
+				if err != nil {
+					return err
+				}
+				cfg.BPFObjBytes = bpfBytes
+			} else {
+				if debug {
+					fmt.Println("BTF is not enabled")
+				}
+				bpfFile, err := getBPFObjectPath()
+				if err != nil {
+					return err
+				}
+				cfg.BPFObjPath = bpfFile
+				bpfBytes, err := ioutil.ReadFile(bpfFile)
+				if err != nil {
+					return err
+				}
+				cfg.BPFObjBytes = bpfBytes
```

Comment:
These four lines are the same for all cases - you can take it out of the if statement